### PR TITLE
Fix CI for internal PRs with CI changes

### PR DESCRIPTION
**Pull request by Pwuts**
### Background
CI currently breaks quite easily because we use `pull_request_target`: https://github.com/Significant-Gravitas/Auto-GPT/actions/runs/5154842393/jobs/9287034366#logs

### Changes
* Use `pull_request` instead of `pull_request_target` for internal pull requests
* Clean up CI workflow
* Fix git authentication set-up

### Test Plan
See checks below: since this is an internal PR, the CI runs on `pull_request`. For PRs from forks, it should still run on `pull_request_target`.

### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -77,25 +125,108 @@ jobs:
...
-          git remote add target https://${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.event.pull_request.base.repo.full_name }}
-          git push -f target cassette-diff-${{ github.event.pull_request.number }}
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          TOKEN=${{ secrets.PAT_REVIEW }}
+          REPO=${{ github.repository }}
+
+          if [[ "${{ steps.push_cassettes.outputs.updated }}" == "true" ]]; then
+            echo "Adding label and comment..."
+            curl -X POST \
+            -H "Authorization: Bearer $TOKEN" \
```
Comment (ntindle) [inline .github/workflows/ci.yml:219]: Don’t save token to a var? Just use it directly? Also does this log?

**Review by ntindle — APPROVED** [review]
nit: move all of these into a scripts folder for ci

**Review by richbeales — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Pwuts) [reply .github/workflows/ci.yml:219]: It's scrubbed from the log because it's a secret

**Review by merwanehamadi — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -77,25 +125,108 @@ jobs:
...
       - name: Upload coverage reports to Codecov
         uses: codecov/codecov-action@v3
 
-      - name: Stage new files and commit
+      - id: setup_git_auth
+        name: Set up git token authentication
+        run: |
+          config_key="http.${{ github.server_url }}/.extraheader"
+          git config "$config_key" \
+            "Authorization: Basic x-access-token:${{ secrets.PAT_REVIEW }}"
```
Comment (waynehamadi) [inline .github/workflows/ci.yml:146]: @Pwuts this is not usable in a pull request event

**Review by merwanehamadi — CHANGES_REQUESTED** [review]

**Review by merwanehamadi — APPROVED** [review]

**Pull request merged**